### PR TITLE
[CAS-802] Allow customization of styles

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-802

### Description

This PR allows to override global styles via `StyleTransformer`. 

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListViewStyle.kt [hunk 1]
```diff
@@ -4,41 +4,46 @@ import android.content.Context
...
-            lastMessageSize = a.getDimensionPixelSize(
-                R.styleable.ChannelListView_streamUiLastMessageTextSize,
-                resources.getDimensionPixelSize(R.dimen.stream_ui_channel_item_message)
-            ).toFloat()
-
-            lastMessageDateTextSize = a.getDimensionPixelSize(
-                R.styleable.ChannelListView_streamUiLastMessageDateTextSize,
-                resources.getDimensionPixelSize(R.dimen.stream_ui_channel_item_message_date)
-            ).toFloat()
+public class ChannelListViewStyle(
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListViewStyle.kt:10]: These constructors are public for Java users, as it is inconvenient to use copy constructors from Java.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarStyle.kt [hunk 2]
```diff
@@ -19,9 +20,8 @@ public data class AvatarStyle(
 ) {
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/avatar/AvatarStyle.kt:21]: A big chunk of work is left here if we want to customize `AvatarView` for each particular view.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListItemStyle.kt [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 
-public class MessageListItemStyle private constructor(
+public class MessageListItemStyle(
     @ColorInt public val messageBackgroundColorMine: Int?,
     @ColorInt public val messageBackgroundColorTheirs: Int?,
     @ColorInt public val messageTextColorMine: Int?,
     @ColorInt public val messageTextColorTheirs: Int?,
     @ColorInt public val messageLinkTextColorMine: Int?,
     @ColorInt public val messageLinkTextColorTheirs: Int?,
     public var reactionsEnabled: Boolean,
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/MessageListItemStyle.kt:16]: This field is for some reason overridden programmatically.

**Comment by bychkovdmitry** [comment]
Possible source of bugs if the following order is not respected:
1) Style attributes are applied
2) `StyleTransformer` transformation is performed (overrides 1)
3) A member function to modify some style value is invoked  (overrides 1, 2)

**Review by ogkuzmin — APPROVED** [review]
I think it's a good step! Nice job 👍

**Pull request merged**